src/Controller/Action/AddWishlistToUserAction.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file was created by developers working at BitBag
+ * Do you need more information about us and what we do? Visit our https://bitbag.io website!
+ * We are hiring developers from all over the world. Join us and start your new, exciting adventure and become part of us: https://bitbag.io/career
+*/
+
+declare(strict_types=1);
+
+namespace BitBag\SyliusWishlistPlugin\Controller\Action;
+
+use BitBag\SyliusWishlistPlugin\Command\Wishlist\AddWishlistToUser;
+use BitBag\SyliusWishlistPlugin\Repository\WishlistRepositoryInterface;
+use Sylius\Component\Core\Model\ShopUserInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
+use Symfony\Component\Messenger\Exception\HandlerFailedException;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Webmozart\Assert\Assert;
+
+final class AddWishlistToUserAction
+{
+    private MessageBusInterface $commandBus;
+
+    private FlashBagInterface $flashBag;
+
+    private TranslatorInterface $translator;
+
+    private WishlistRepositoryInterface $wishlistRepository;
+
+    private UrlGeneratorInterface $urlGenerator;
+
+    private TokenStorageInterface $tokenStorage;
+
+    public function __construct(
+        MessageBusInterface $commandBus,
+        FlashBagInterface $flashBag,
+        TranslatorInterface $translator,
+        WishlistRepositoryInterface $wishlistRepository,
+        UrlGeneratorInterface $urlGenerator,
+        TokenStorageInterface $tokenStorage
+    ) {
+        $this->commandBus = $commandBus;
+        $this->flashBag = $flashBag;
+        $this->translator = $translator;
+        $this->wishlistRepository = $wishlistRepository;
+        $this->urlGenerator = $urlGenerator;
+        $this->tokenStorage = $tokenStorage;
+    }
+
+    public function __invoke(Request $request): Response
+    {
+        $wishlistId = $request->attributes->getInt('id');
+        $wishlist = $this->wishlistRepository->find($wishlistId);
+
+        /** @var ShopUserInterface $shopUser */
+        $shopUser = $this->tokenStorage->getToken()->getUser();
```

Comment:
are you sure the token always is not null ?

Reply:
can be moved to definition block

```
        /** @var ShopUserInterface $shopUser */
        $shopUser = $this->tokenStorage->getToken()->getUser();
        $wishlistId = $request->attributes->getInt('id');
        $wishlist = $this->wishlistRepository->find($wishlistId);

```

Reply:
Done.